Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -235,19 +231,23 @@
 % end
 
 %% 5. Decode the Hadamard data
-decodedPWI4D = zeros(size(imEncoded,1), size(imEncoded,2), size(imEncoded,3), nDecodedVolumes * nRepetitions);    
-idx=0;
-for iRepetition = 1:nRepetitions
-    for iTE = 1:xQ.NumberEchoTimes
-        for iTI = 1:nDecodedTI
-            
-            indexPositive = find(xQ.TimeEncodedMatrix(iTI,:)==1);
-            indexNegative = find(xQ.TimeEncodedMatrix(iTI,:)==-1);
-            decodedPWI4D(:,:,:,((iTE-1)*nDecodedTI+iTI)+(iRepetition-1)*nDecodedVolumes) = sum(imEncoded(:,:,:,(indexPositive+idx)),4) - sum(imEncoded(:,:,:,(indexNegative+idx)),4);
-            
-        end
-        idx = idx+xQ.TimeEncodedMatrixSize;
-    end
+% We assume that the entire encoded volume is reorder into several repetitions of basic Hadamard blocks (repetitions or TE are counted the same)
+decodedPWI4D = zeros(size(imEncoded,1), size(imEncoded,2), size(imEncoded,3), nEncodedVolumes / xQ.TimeEncodedMatrixSize * (xQ.TimeEncodedMatrixSize-1)); 
+decodedBlock = zeros(size(imEncoded,1), size(imEncoded,2), size(imEncoded,3), xQ.TimeEncodedMatrixSize-1); % Temporary single decoded Hadamard block
```

Comment:
No. Block isn't jargon, block means a different thing - it doesn't make it less accessible, it is about calling things with the correct names. Hadamard ASL acquisition is composed of several repetitions and each repetition of several blocks. So the correct name would be decodedBlocks. In theory we could call it both decodedBlock or decodedRepetition and always find an argument why this name is better. But I would like to keep the "block" to make it clear that we are down to the atomic hadamard unit. Because repetition includes several TEs. Block doesn't.

Reply:
"Block" is a standard term used in all Hadamard papers to refer to a single volume in the Hadamard time series.

You have to complain with Matthias Gunther that he calls it a "block".

Reply:
You don't answer my question :) The lexicon has more examples where they use something different than arbitrarily introduced by a PI. So is it an arbitrary term? Or is there a reason for this term? I want to understand it ;)

Again, my question: 1. can you describe how this would be as definition in BIDS? The description of the block? As specific as possible, so I understand it? And 2. it is not a Hadamard mathematical term right, it is something that Matthias came up with?

Reply:
I wish I knew what answer would make you happy from the start :D

No worries - glad we have a consensus on this. You can resolve.